tirocks/src/encryption.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::fmt::{self, Debug, Formatter};
+use std::{mem, ptr};
+
+use crate::util::utf8_name;
+use crate::{Code, Result, Status};
+use libc::c_void;
+use tirocks_sys::{
+    crocksdb_file_encryption_info_t, rocksdb_Slice, rocksdb_Status,
+    rocksdb_encryption_EncryptionMethod, rocksdb_encryption_KeyManager,
+};
+
+pub type EncryptionMethod = rocksdb_encryption_EncryptionMethod;
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct FileEncryptionInfo {
+    method: EncryptionMethod,
+    key: Vec<u8>,
+    iv: Vec<u8>,
+}
+
+impl FileEncryptionInfo {
+    #[inline]
+    pub fn new(method: EncryptionMethod, key: Vec<u8>, iv: Vec<u8>) -> FileEncryptionInfo {
+        FileEncryptionInfo { method, key, iv }
+    }
+
+    #[inline]
+    pub fn method(&self) -> EncryptionMethod {
+        self.method
+    }
+
+    #[inline]
+    pub fn key(&self) -> &[u8] {
+        &self.key
+    }
+
+    #[inline]
+    pub fn iv(&self) -> &[u8] {
+        &self.iv
+    }
+
+    unsafe fn copy_to(&self, file_info: *mut crocksdb_file_encryption_info_t) {
+        let info = &mut *file_info;
+        info.method = self.method;
+        info.key = self.key.as_ptr() as _;
```

Comment:
Not exactly "copy". I kind of prefer using `crocksdb_file_encryption_info_set_key`.